src/control/lib/ui/list_flags.go

```diff
@@ -71,3 +77,71 @@ func (f *HostSetFlag) UnmarshalFlag(fv string) error {
 func (f *HostSetFlag) MarshalJSON() ([]byte, error) {
 	return []byte(f.String()), nil
 }
+
+// MemberStateSetFlag is a go-flags compatible flag type for handling inputs that can be converted
+// to a system.MemberState slice.
+type MemberStateSetFlag struct {
+	States system.MemberState
+}
+
+// Empty returns true if the flag was not set.
+func (f *MemberStateSetFlag) Empty() bool {
+	return f.States == 0
+}
+
+// States are specified as a comma separated list of AwaitFormat Starting Ready Joined Stopping
+// Stopped Excluded AdminExcluded Errored Unresponsive
+func memberStateMaskFromStrings(statesStr string) (system.MemberState, error) {
+	var states []system.MemberState
+
+	for _, tok := range strings.Split(statesStr, ",") {
+		ms := system.MemberStateFromString(strings.TrimSpace(tok))
+		if ms == system.MemberStateUnknown {
+			return 0, errors.Errorf("invalid state name %q", tok)
+		}
+		states = append(states, ms)
+	}
+
+	mask, _ := system.MemberStates2Mask(states...)
```

Comment:
minor - could just build the state mask in the loop, since we're not really using any of the extra functionality in the MemberStates2Mask function

Reply:
done